### PR TITLE
Update py-backports-functools-lru-cache to 1.4

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-backports-functools-lru-cache/package.py [hunk 1]
```diff
@@ -28,9 +28,10 @@
 class PyBackportsFunctoolsLruCache(PythonPackage):
     """Backport of functools.lru_cache from Python 3.3"""
 
-    homepage = "https://bitbucket.org/jaraco/backports.functools_lru_cache"
-    url      = "https://pypi.io/packages/source/b/backports.functools_lru_cache/backports.functools_lru_cache-1.0.1.zip"
+    homepage = "https://github.com/jaraco/backports.functools_lru_cache"
+    url = "https://pypi.io/packages/source/b/backports.functools_lru_cache/backports.functools_lru_cache-1.4.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-backports-functools-lru-cache/package.py]: If you change from zip to tar.gz you need to make sure the old checksum is updated or the old version downloads from a different URL.

**Comment by adamjstewart** [comment]
I don't seem to be able to restart the Travis CI builds for this PR. I'm going to open up an issue with Travis.

**Comment by adamjstewart** [comment]
Jk, it took 5 minutes for the jobs to actually restart. I've never seen that happen before...

**Comment by healther** [comment]
Can fetch manually, but Travis failed. Is https://pypi.io having problems? https://status.python.org/#month Doesn't seem to show abnormal error rates. @adamjstewart should I retrigger Travis?

**Pull request merged**